www/assets/js/subnav.js

```diff
@@ -1,18 +1,36 @@
 const initSubNav = () => {
-  $(".navbar .nav-link").on("click", e => {
+  // [START] Scrolling to appropriate section when subnav item is clicked
+  $(".on-page-sub-nav .nav-link").on("click", e => {
     const navbar = $(".navbar")
     const clickedLink = $(e.target)
     navbar.find(".active").removeClass("active")
     clickedLink.addClass("active")
     $(".navbar-current-active").text(clickedLink.text())
     const target = $(clickedLink.attr("href")).get(0)
-    const yOffset = -navbar.height()
+    const yOffset = -(navbar.height() + 20)
     const y = target.getBoundingClientRect().top + window.scrollY + yOffset
     window.scrollTo({
       top:      y,
       behavior: "smooth"
     })
   })
+  // [END] Scrolling to appropriate section when subnav item is clicked
+
+  // [START] highlight subnav item as user scrolls
+  $(window).on("scroll", function() {
+    const position = window.pageYOffset
+    $(".section").each(function() {
+      const target = $(this).offset().top
+      const navHeight = $(".on-page-sub-nav").height()
+      if (position >= target - navHeight - 50) {
+        const navLinks = $(".on-page-sub-nav  li a")
+        const id = $(this).attr("id")
+        navLinks.removeClass("active")
+        $(`.on-page-sub-nav  li a[href="#${id}"]`).addClass("active")
```

Comment:
This code currently adds active class to the suitable nav-item on scroll. However if you view this in mobile view the nav-brand does not change to the suitable  nav-item text on scroll, it remains the default one or the first one.However on opening the nav menu and clicking nav-item it doesn't change the nav-brand

```
const activeElement = $(`.on-page-sub-nav  li a[href="#${id}"]`)
activeElement.addClass("active")
const brandText = activeElement.text()
$(".navbar-brand > span").text(brandText)
```

The above code can resolve this issue after that feel free to merge

Reply:
Thanks for pointing it out Wassaf!
I've changed it, please have a look.